pallets/creditcoin/src/helpers.rs

```diff
@@ -144,9 +204,44 @@ impl<T: Config> Pallet<T> {
 
 		let block = Self::block_number();
 
+		DealOrders::<T>::mutate(
+			deal_order_id.expiration(),
+			&deal_order_id.hash(),
+			|deal_order| -> Result<(), crate::Error<T>> {
+				let mut deal_order = deal_order.as_mut().ok_or(Error::<T>::NonExistentDealOrder)?;
+				let currency = match &from.blockchain {
```

Comment:
Because this is the legacy case, where loan terms only have a placeholder currency, so we need to compute the currency from the blockchain + transfer kind, and then write it to the loan terms.